### PR TITLE
feat(autoware_auto_perception_rviz_plugin): rviz predicted objects multi-threading

### PR DESCRIPTION
**Pull request by tkimura4**
## Description
When the number of predicted objects is large, Rviz's Hz drops significantly and drawing stops, so I multithreaded the visualization of predicted objects.

image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp [hunk 1]
```diff
@@ -127,6 +127,11 @@ class AUTOWARE_AUTO_PERCEPTION_RVIZ_PLUGIN_PUBLIC ObjectPolygonDisplayBase
     m_marker_common.addMessage(marker_ptr);
   }
 
+  void add_marker(visualization_msgs::msg::MarkerArray::ConstSharedPtr markers_ptr)
```
Comment (takayuki5168) [inline common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp:130]: @tkimura4 
Did you measure how much the visualization delay was shortened?
I'm just curious

**Review by taikitanaka3 — APPROVED** [review]
LGTM
drawing becomes much bettter
image

File: common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp (hunk 1 above)
Comment (tkimura4) [reply common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp:130]: On my Desktop PC, The hz of rviz visualization does not decrease even when a number of objects exceeds 50.
So, I cannot measure the improvement of visualization delay.

Comment (takayuki5168) [reply common/autoware_auto_perception_rviz_plugin/include/object_detection/object_polygon_display_base.hpp:130]: Cool, thanks!

**Review by taichiH — APPROVED** [review]

**Pull request merged**